clash-lib/src/Clash/Backend/Verilog.hs

```diff
@@ -1216,7 +1216,7 @@ exprLitV = exprLit undefValue
 exprLit :: Lens' s (Maybe (Maybe Int)) -> Maybe (HWType,Size) -> Literal -> Ap (State s) Doc
 exprLit _ Nothing (NumLit i) = integer i
 
-exprLit k (Just (hty,sz)) (NumLit i) = case hty of
+exprLit k (Just (hty,sz)) (NumLit i0) = case hty of
```

Comment:
This code doesn't do the same as the VHDL code. In the VHDL code the comparisons would be against this `i0` to determine whether to create a bitstring or an integer literal.

Reply:
No it doesn't do the same as VHDL, because it doesn't have to.

In the VHDL backend we generate three forms of (signed) number literals:
```vhdl
form1 <= to_signed(2147483647,66);
form2 <= signed'(x"00000000080000000");
form3 <= signed'("000000000000000000000000000000000010000000000000000000000000000000");
```
And clash prefers the first form because it's generally more readable, but when the value gets outside of the range of integer that doesn't work anymore.

But in Verilog there isn't this size limit, clash can just generate `66'sd2147483648`.

The code that I copied over from the VHDL backend changes this:
```
8'sd255
```
to this:
```
-8'sd1;
```

Reply:
Sorry, bad example, verilator still accepts that ^

But this:
```
  assign result = 8'sd1000;
```
results in:
```
%Warning-WIDTHTRUNC: Blaat.topEntity/topEntity.v:14:19: Value too large for 8 bit number: 1000
   14 |   assign result = 8'sd1000;
      |                   ^~~~~~~~
                     ... For warning description see https://verilator.org/warn/WIDTHTRUNC?v=5.018
                     ... Use "/* verilator lint_off WIDTHTRUNC */" and lint_on around source to disable this message.
%Error: Exiting due to 1 warning(s)
```